### ISSUE TITLE
Macro fails to resolve Sets with case objects

### ISSUE THREAD

**Issue by bomgar** [adamw/macwire#103]
```scala
class WireApp extends App {
  import com.softwaremill.macwire._
  trait TestTrait
  case object TestObject1 extends TestTrait
  case object TestObject2 extends TestTrait
  case class Test(set: Set[TestTrait])

  val set: Set[TestTrait] = Set(TestObject2, TestObject1)
  // this line is needed for the macro to compile
  // val set2 = set
  wire[Test] // Cannot find a value of type: [Set[WireApp.this.TestTrait]]
}
```

**Comment by adamw** [comment]
Note that this compiles:

```
class Test(set: Set[String])
  val set: Set[String] = Set[String]()
  wire[Test]
```

while this doesn't:

```
class Test(set: Set[String])
  val set: Set[String] = Set()
  wire[Test]
```

**Comment by adamw** [comment]
Maybe that's somehow connectd to #102?

**Comment by adamw** [comment]
(though if anything, #102 should fix the problem ;) )

**Comment by bomgar** [comment]
I upgraded to 2.2.4 but still have the same problem.

**Comment by adamw** [comment]
Even more interestingly this works:

```
class Test(set: Set[String])
  val set = Set[String]()
  wire[Test]
```

But this doesn't:

```
class Test(set: Set[String])
val set: Set[String] = Set()
wire[Test]
```

(which would seem, is the easier case). @mkubala maybe any ideas? ;)

@bomgar so as a work-around, this works:

```
trait TestTrait
  case object TestObject1 extends TestTrait
  case object TestObject2 extends TestTrait
  case class Test(set: Set[TestTrait])

  val set = Set[TestTrait](TestObject2, TestObject1)
  // this line is needed for the macro to compile
  // val set2 = set
  wire[Test] // Cannot find a value of type: [Set[WireApp.this.TestTrait]]
```

(but obviously both variants should)

**Comment by adamw** [comment]
Released as 2.2.5, thanks @tkroman !
